test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should not propagate errors for Multiplexed Session when default credentials not set', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        message: 'Could not load the default credentials',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should not propagate errors for Multiplexed Session when projectId not set', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        code: Status.NOT_FOUND,
+        message: 'Unable to detect a Project Id in the current environment',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should propagate errors for Multiplexed Session which gets emitted', async () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const fakeError = new Error();
+      sandbox.stub(multiplexedSession, '_createSession').rejects(fakeError);
+      const errorPromise = new Promise<void>((resolve, reject) => {
+        multiplexedSession.once('error', err => {
+          if (err === fakeError) {
+            resolve();
+          } else {
+            reject(new Error('Unexpected error emitted'));
+          }
+        });
+      });
+
+      multiplexedSession.createSession();
+
+      await errorPromise;
+    });
+  });
+
+  describe('_maintain', () => {
+    it('should set an interval to refresh mux sessions', done => {
+      const expectedInterval =
+        multiplexedSession.refreshRate! * 24 * 60 * 60000;
+      const clock = sandbox.useFakeTimers();
+
+      const _refreshStub = sandbox
+        .stub(multiplexedSession, '_refresh')
+        .callsFake(async () => done());
+
+      multiplexedSession._maintain();
+      clock.tick(expectedInterval);
+      assert.strictEqual(_refreshStub.callCount, 1);
+    });
+  });
+
+  describe('_createSession', () => {
+    it('should create the mux sessions', async () => {
+      await multiplexedSession._createSession();
+      assert.strictEqual(createSessionStub.callCount, 1);
+    });
+
+    it('should pass the multiplexed option', async () => {
+      const multiplexed = {multiplexed: true};
+      await multiplexedSession._createSession();
+      assert.deepStrictEqual(createSessionStub.lastCall.args[0], multiplexed);
+    });
+
+    it('should not make multiple requests', async () => {
+      createSessionStub.onCall(0).resolves([[{}, {}]]);
+      createSessionStub.onCall(1).resolves([[{}]]);
+
+      await multiplexedSession._createSession();
```

Comment:
Did you mean to make multiple calls to `_createSession` here ?

Reply:
done

Reply:
there some modification needed in the test, I have modified it.
No, I mean the multiple calls to createSession method of the database class here, since _createSession is calling that method

Reply:
this test is removed based on discussion